src/main/java/appeng/client/gui/implementations/GuiMEMonitorable.java

```diff
@@ -290,8 +295,10 @@ public void initGui()
 		offset += 20;
 
 		this.buttonList.add(
-				this.searchBoxSettings = new GuiImgButton( this.guiLeft - 18, offset, Settings.SEARCH_MODE, AEConfig.instance().getConfigManager().getSetting(
-						Settings.SEARCH_MODE ) ) );
+				this.searchBoxSettings = new GuiImgButton( this.guiLeft - 18, offset, Settings.SEARCH_MODE, AEConfig.instance()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)